### ISSUE TITLE
Add a message on a critical hit

### ISSUE THREAD

**Issue by kaelad02** [kaelad02/adv-reminder#29]
Define a new flag, `flags.adv-reminder.message.damage.critical`, that shows messages when a critical hit happens. Would be useful for some of the new feats like Crusher that can do something extra if a critical hit occurs.

Right now, the only way to detect that a critical hit occurred is either with flags for certain statuses or if using Midi and it detected the crit. The way things are ordered in Advantage Reminder is the flags are done after messages, making the first one harder. Nothing that can't be switched though. The tougher problem is that Advantage Reminder runs before Midi so it won't know that a nat 20 was rolled since Midi would set the critical flag for that. There's an upcoming system feature that adds a bunch of hooks that I might switch over to. That would make this possible, assuming Midi will call those hooks too when it overrides the `rollDamage` function.

**Comment by kaelad02** [comment]
added code attempting this to the `critical-message` branch

**Comment by YenBenGrey** [comment]
Hello there, I have just started not using MidiQol and just wanted to know if this was a thing that can be used without MidiQol?
Would you have to put the flag above in every weapon and spell that attacks and does damage?

**Comment by kaelad02** [comment]
No, that "flags for certain statuses" was more about the `critical` option that's passed in the `rollDamage` function, not a flag on the items themselves. The main issue is AR cannot tell that it's a critical hit or not when the dialog is presented, even with Midi b/c of the order. I tried to implement it but struck out and this is the reminder of that. If things change I can revisit it, but it's still the case.
